VideoLocker/src/main/java/org/edx/mobile/view/UserProfileActivity.java

```diff
@@ -0,0 +1,48 @@
+package org.edx.mobile.view;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.Fragment;
+
+import com.google.inject.Inject;
+
+import org.edx.mobile.R;
+import org.edx.mobile.base.BaseSingleFragmentActivity;
+
+import roboguice.inject.InjectExtra;
+
+public class UserProfileActivity extends BaseSingleFragmentActivity {
+    public static final String EXTRA_USERNAME = "username";
+
+    public static Intent newIntent(@NonNull Context context, @NonNull String username) {
+        return new Intent(context, UserProfileActivity.class)
+                .putExtra(EXTRA_USERNAME, username);
+    }
+
+    @Inject
+    private UserProfileFragment fragment;
```

Comment:
Agreed, this pattern came from cloning another activity, not sure why we are doing it anywhere.

(Fixed)